ludwig/schema/hyperopt/hyperopt.py

```diff
@@ -0,0 +1,97 @@
+from abc import ABC
+from typing import Any
+
+from marshmallow_dataclass import dataclass
+
+from ludwig.constants import LOSS
+from ludwig.features.feature_registries import output_type_registry
+from ludwig.schema import utils as schema_utils
+from ludwig.schema.hyperopt.executor import ExecutorConfig, ExecutorDataclassField
+from ludwig.schema.hyperopt.search_algorithm import SearchAlgorithmConfig, SearchAlgorithmDataclassField
+
+# hyperopt:
+#   search_alg:
+#     type: hyperopt
+#     random_state_seed: 42
+#   executor:
+#     type: ray
+#     num_samples: 10
+#     time_budget_s: 3600
+#     scheduler:
+#       type: async_hyperband
+#       time_attr: time_total_s
+#       max_t: 3600
+#       grace_period: 72
+#       reduction_factor: 5
+#     cpu_resources_per_trial: 1
+#   parameters:
+#     trainer.learning_rate:
+#       space: choice
+#       categories:
+#         - 0.005
+#         - 0.01
+#         - 0.02
+#         - 0.025
+#   output_feature: combined
+#   goal: minimize
+#   metric: loss
+
+
+def get_hyperopt_metric_options():
+    all_metrics = []
+    for oftype in output_type_registry:
+        ofcls = output_type_registry[oftype]
+        all_metrics += ofcls.metric_functions.keys()
+    return all_metrics
+
+
+@dataclass
+class HyperoptConfig(schema_utils.BaseMarshmallowConfig, ABC):
+    """Basic hyperopt settings."""
+
+    output_feature: str = "combined"  # TODO: make more restrictive
```

Comment:
Not sure if there's a way to specify that this can only be combined, or one of the output feature types in the schema/config. But this is the correct default 👍

Reply:
@arnavgarg1 yup that is the intention with the TODO. It's possible but it'll require some refactoring worth a different PR